app/uk/gov/hmrc/agentassurance/controllers/PropertiesController.scala

```diff
@@ -16,57 +16,32 @@
 
 package uk.gov.hmrc.agentassurance.controllers
 
-
-import play.api.libs.json.Json
 import play.api.mvc.{Request, Result}
 import uk.gov.hmrc.agentassurance.repositories.PropertiesRepository
 import uk.gov.hmrc.play.bootstrap.controller.BaseController
 import uk.gov.hmrc.agentassurance.model._
-
 import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.Future
 
-abstract class PropertiesController (repository: PropertiesRepository) extends BaseController {
+abstract class PropertiesController(repository: PropertiesRepository) extends BaseController {
 
   def key: String
 
+  protected def getCollectionUtrs(key: String)(implicit request: Request[Any]): Future[Result] = {
+    repository.getUtrsForKey(key).map(response => if(response.nonEmpty) Ok(response.mkString(",")) else NoContent)
+  }
+
   protected def baseCreateProperty(value: Value)(implicit request: Request[Any]) = {
-    repository.findProperty(key).flatMap { op =>
-      if (op.isEmpty) {
-        repository.createProperty(value.toProperty(key)).map(_ => Created)
-      } else {
-        Future successful Conflict(Json.toJson(ErrorBody("PROPERTY_EXISTS", "Property already exists")))
-      }
-    }
+      repository.updateProperty(value.toProperty(key)).map(response => if (response) Created else Conflict)
```

Comment:
I think we should return same error messages as before.

Reply:
Yeah, was thinking about that, something to discuss I suppose